include/swift/AST/DiagnosticEngine.h

```diff
@@ -629,7 +629,27 @@ namespace swift {
     DiagnosticState(DiagnosticState &&) = default;
     DiagnosticState &operator=(DiagnosticState &&) = default;
   };
-    
+
+  class LocalizationProducer {
```

Comment:
We might want to move this class  and `YAMLLocalizationProducer` to `LocalizationFormat.h` and place them into `swift::diag` namespace.